### PR TITLE
feat(material): Adds reverse translators for window gas materials

### PR DESCRIPTION
**Pull request by tanushree04**
Adds reverse translator and tests for window gas, window gas mixture, window gas custom.

### CODE REVIEW

File: lib/from_openstudio/material/window_gas_mixture.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        # set hash values from OpenStudio Object
+        hash[:identifier] = material.nameString
+        hash[:thickness] = material.thickness
+        hash[:gas_types] = []
+        hash[:gas_fractions] = []
+        hash[:gas_types] << material.gas1Type
+        hash[:gas_fractions] << material.gas1Fraction
+        number_of_gases = material.numberofGasesinMixture
+        # Check and add if more than 1 gas is present
+        if number_of_gases > 1
```
Comment (chriswmackey) [inline lib/from_openstudio/material/window_gas_mixture.rb:51]: The honeybee `EnergyWindowMaterialGasMixture` always has to have between 2 and 4 gasses. You can see this in the schema docs:
https://www.ladybug.tools/honeybee-schema/model.html#tag/energywindowmaterialgasmixture_model

So you should only be using this translation pathway if there are 2 or more gasses.

Also all of these `if` statements could be written a lot more cleanly with a single for loop like so:
```
(1..number_of_gases).each do |n|
    hash[:gas_types] << material.send('gas' + n.to_s + 'Type')
end
```
(I think that's the correct equivalent of [getting an attribute by its text name in Ruby](https://stackoverflow.com/questions/786412/what-is-the-ruby-equivalent-of-pythons-getattr))

File: lib/from_openstudio/model.rb [hunk 2]
```diff
@@ -150,6 +153,20 @@ def self.materials_from_model(openstudio_model)
       openstudio_model.getBlinds.each do |material|
         result << EnergyWindowMaterialBlind.from_material(material)
       end
+      openstudio_model.getGass.each do |material|
```
Comment (chriswmackey) [inline lib/from_openstudio/model.rb:156]: Interesting. Am I correct that this method won't return any `GasMixture` materials?

**Review by chriswmackey — COMMENTED** [review]
Thanks, @tanushree04 . It's looking good but there are a few tweaks to make before we merge.

File: lib/from_openstudio/model.rb (hunk 2 above)
Comment (tanushree04) [reply lib/from_openstudio/model.rb:156]: Yes thats correct, Gas and GasMaterial are separate in OS.

File: lib/from_openstudio/material/window_gas_mixture.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        hash[:gas_types] = []
+        hash[:gas_fractions] = []
+        number_of_gases = material.numberofGasesinMixture
+        if number_of_gases > 1
+            (1..number_of_gases).each do |n|
+                hash[:gas_types] << material.send('gas' + n.to_s + 'Type')
+            end
+            hash[:gas_fractions] << material.gas1Fraction
+            hash[:gas_fractions] << material.gas2Fraction
+            if number_of_gases > 2
```
Comment (chriswmackey) [inline lib/from_openstudio/material/window_gas_mixture.rb:54]: Do you plan to use the same strategy for `gas_fractions` as you do for `gas_types`?  There's really no need for all of these `if` statements when a single loop will do.

Comment (tanushree04) [reply lib/from_openstudio/material/window_gas_mixture.rb:54]: For the `gas3Fraction` and `gas4Fraction` are boost optionals in OpenStudio, so I will have to use the .get method on them.

Comment (chriswmackey) [reply lib/from_openstudio/material/window_gas_mixture.rb:54]: Thanks for clarifying. I guess this is fine, then. It doesn't feel very elegant but I guess it's not our fault and we can blame the EnergyPlus IDD combined with some of the idiosyncrasy of boost.

**Review by chriswmackey — APPROVED** [review]
Looks good, @tanushree04 . Let's merge!

**Pull request merged**